File: Dished/Layout/DAReviewDetailCollectionViewCell.m
//
//  DAReviewDetailCommentCollectionViewCell.m
//  Dished
//
//  Created by POST on 9/8/14.
//  Copyright (c) 2014 Dished. All rights reserved.
//

#import "DAReviewDetailCollectionViewCell.h"


@interface DAReviewDetailCollectionViewCell() <DALinkedTextViewDelegate>

@end


@implementation DAReviewDetailCollectionViewCell

+ (DAReviewDetailCollectionViewCell *)sizingCell
{
    NSString *nibName = NSStringFromClass( [DAReviewDetailCollectionViewCell class] );
    DAReviewDetailCollectionViewCell *sizeCell = [[[NSBundle mainBundle] loadNibNamed:nibName owner:self options:nil] lastObject];

    return sizeCell;
}

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    self.textView.scrollsToTop = NO;
    self.textView.textContainerInset = UIEdgeInsetsZero;
    self.textView.tapDelegate = self;
    
    self.opaque = YES;
}

- (void)linkedTextView:(DALinkedTextView *)textView tappedOnText:(NSString *)text withLinkedTextType:(eLinkedTextType)textType
{
    if( [self.delegate respondsToSelector:@selector(textViewTappedOnText:withTextType:inCell:)] )
    {
        [self.delegate textViewTappedOnText:text withTextType:textType inCell:self];
    }
}

- (void)prepareForReuse
{
    [super prepareForReuse];
    
    self.iconImageView.hidden = NO;
    self.iconImageView.image = nil;
    self.textView.text = nil;
    self.textView.attributedText = nil;
}

@end

File: Dished/Layout/DAReviewDetailCollectionViewCell.swift
//
//  DAReviewDetailCollectionViewCell.swift
//  Dished
//
//  Created by Ryan Khalili on 12/21/14.
//  Copyright (c) 2014 Dished. All rights reserved.
//

import UIKit

@objc protocol DAReviewDetailCollectionViewCellDelegate {
    optional func textViewTappedOnText( text: String, withTextType textType: eLinkedTextType, inCell cell: DAReviewDetailCollectionViewCell )
}

class DAReviewDetailCollectionViewCell: DACollectionViewCell, DALinkedTextViewDelegate {
    
    weak var delegate: DAReviewDetailCollectionViewCellDelegate?
    
    var iconImageView: UIImageView!
    var textView: DALinkedTextView!
    
    class func sizingCell() -> DAReviewDetailCollectionViewCell {
        let cell = DAReviewDetailCollectionViewCell(frame: CGRectMake(0.0, 0.0, 320.0, 50.0))
        cell.layoutIfNeeded()
        return cell
    }
    
    func linkedTextView( textView: DALinkedTextView!, tappedOnText text: String!, withLinkedTextType textType: eLinkedTextType ) {
        if text != nil {
            delegate?.textViewTappedOnText?( text, withTextType: textType, inCell: self )
        }
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        
        iconImageView.hidden = false
        iconImageView.image = nil
        textView.text = nil
        textView.attributedText = nil
    }
    
    override func setupViews() {
        iconImageView = UIImageView()
        iconImageView.contentMode = UIViewContentMode.ScaleAspectFill
        iconImageView.clipsToBounds = true
        addSubview(iconImageView)
        iconImageView.autoPinEdgeToSuperviewEdge(ALEdge.Left, withInset: 9.0)
        iconImageView.autoPinEdgeToSuperviewEdge(ALEdge.Top, withInset: 4.0)
        iconImageView.autoSetDimensionsToSize(CGSizeMake(12.0, 11.0))
        
        textView = DALinkedTextView()
        textView.font = DAConstants.primaryFontWithSize(14.0)
        textView.selectable = false
        textView.editable = false
        textView.scrollsToTop = false
        textView.textContainerInset = UIEdgeInsetsZero
        textView.tapDelegate = self
        textView.backgroundColor = UIColor.clearColor()
        addSubview(textView)
        textView.autoPinEdgeToSuperviewEdge(ALEdge.Top)
        textView.autoPinEdgeToSuperviewEdge(ALEdge.Bottom)
        textView.autoPinEdgeToSuperviewEdge(ALEdge.Right, withInset: 15.0)
        textView.autoPinEdge(ALEdge.Left, toEdge: ALEdge.Right, ofView: iconImageView, withOffset: 2.0)
    }
}